header-rewrite-filter/utility.cc

```diff
@@ -46,6 +46,27 @@ BooleanOperatorType StringToBooleanOperatorType(absl::string_view bool_operator)
     }
 }
 
+FunctionType StringToFunctionType(absl::string_view function) {
+    if (function.find(DYNAMIC_VALUE_HDR) == 0) {
+        return FunctionType::GetHdr;
+    } else if (function.find(DYNAMIC_VALUE_URL_PARAM) == 0) {
+        return FunctionType::Urlp;
```

Comment:
These are searching the full string, can use `string::compare` or invert the find to only search as far as necessary. See https://github.com/DataDog/envoy-filter-example/pull/25/files#r1281672769

Reply:
This is also searching prefixes, so a function `urlpfoobar` is going to match `urlp`. I think it needs to `function == DYNAMIC_VALUE_HDR`